model/src/application/component/button.rs

```diff
@@ -43,34 +42,62 @@ pub struct Button {
 /// Refer to [the Discord Docs/Button Object] for additional information.
 ///
 /// [the Discord Docs/Button Object]: https://discord.com/developers/docs/interactions/message-components#button-object-button-styles
-#[derive(Clone, Copy, Debug, Deserialize_repr, Eq, Hash, PartialEq, PartialOrd, Serialize_repr)]
-#[repr(u8)]
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, PartialOrd, Serialize)]
+#[serde(from = "u8", into = "u8")]
 pub enum ButtonStyle {
     /// Button indicates a primary action.
     ///
     /// Selecting this button style requires specifying the
     /// [`Button::custom_id`] field.
-    Primary = 1,
+    Primary,
     /// Button indicates a secondary action.
     ///
     /// Selecting this button style requires specifying the
     /// [`Button::custom_id`] field.
-    Secondary = 2,
+    Secondary,
     /// Button indicates a successful action.
     ///
     /// Selecting this button style requires specifying the
     /// [`Button::custom_id`] field.
-    Success = 3,
+    Success,
     /// Button indicates a dangerous action.
     ///
     /// Selecting this button style requires specifying the
     /// [`Button::custom_id`] field.
-    Danger = 4,
+    Danger,
     /// Button indicates an action with a link.
     ///
     /// Selecting this button style requires specifying the [`Button::url`]
     /// field.
-    Link = 5,
+    Link,
+    /// Button has an unknown style not known to the library
```

Comment:
Probably good to have some sort of standard for these. We could go with the more technical form of:
```suggestion
    /// Variant value is not registered by the library.
```
or perhaps a more concise and literal explanation is best:
```suggestion
    /// Variant value is unknown to the library.
```